pkg/mimirtool/commands/promql.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package commands
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+type PromQLCommand struct {
+	query       string
+	prettyPrint bool
+}
+
+func (c *PromQLCommand) Register(app *kingpin.Application, _ EnvVarNames) {
+	promqlCmd := app.Command("promql", "PromQL formatting and editing for Grafana Mimir.")
+
+	// PromQL Format Query Command
+	promqlFormatCmd := promqlCmd.Command("format", "Format PromQL query with Prometheus' string formatter; wrap query in quotes for CLI parsing.").Action(c.formatQuery)
+	promqlFormatCmd.Flag("pretty", "use Prometheus' pretty-print formatter").BoolVar(&c.prettyPrint)
```

Comment:
Is Prometheus relevant here? I think we can just say something like "pretty-print expression"

Reply:
I think the fact that it's using upstream code is relevant for a user technical enough to be doing this. It implies a level of stability and consistency with other tooling like `promtool` - using the `--pretty` flag here is identical to the promtool command. The promtool version just doesn't allow non-pretty at this point.

Reply:
I'm not sure that the fact we're using the upstream code is so relevant here or in the docs above - query blocking is a Mimir-only feature, so alignment with how Prometheus formats queries isn't so important, but consistency with how the query-blocking feature formats queries is important.